### ISSUE TITLE
storm 1.0.1 / stable macOS build error

### ISSUE THREAD

**Issue by tjanson** [moves-rwth/storm#6]
```
$ ninja
[95/1187] Performing download step (git clone) for 'l3pp_ext'
Cloning into 'l3pp'...
Note: checking out 'e4f8d7fe6c328849aff34d2dfd6fd592c14070d5'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at e4f8d7f... Fix some references to log4carl
[263/1187] Performing configure step for 'sylvan'
-- sylvan configure command succeeded.  See also /Users/macbook/Code/i2/storm/build/sylvan/src/sylvan-stamp/sylvan-configure-*.log
[383/1187] Performing configure step for 'googletest'
-- googletest configure command succeeded.  See also /Users/macbook/Code/i2/storm/build/googletest-prefix/src/googletest-stamp/googletest-configure-*.log
[591/1187] Performing build step for 'googletest'
-- googletest build command succeeded.  See also /Users/macbook/Code/i2/storm/build/googletest-prefix/src/googletest-stamp/googletest-build-*.log
[595/1187] Performing build step for 'sylvan'
-- sylvan build command succeeded.  See also /Users/macbook/Code/i2/storm/build/sylvan/src/sylvan-stamp/sylvan-build-*.log
[598/1187] Performing configure step for 'cudd3'
-- cudd3 configure command succeeded.  See also /Users/macbook/Code/i2/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-configure-*.log
[599/1187] Performing build step for 'cudd3'
-- cudd3 build command succeeded.  See also /Users/macbook/Code/i2/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-build-*.log
[600/1187] Performing install step for 'cudd3'
-- cudd3 install command succeeded.  See also /Users/macbook/Code/i2/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-install-*.log
[641/1187] Building CXX object src/storm-pgcl/CMakeFiles/storm-pgcl.dir/builder/JaniProgramGraphBuilder.cpp.o
In file included from ../src/storm-pgcl/builder/JaniProgramGraphBuilder.cpp:1:
In file included from ../src/storm-pgcl/builder/JaniProgramGraphBuilder.h:11:
../src/storm-pgcl/storage/ppg/ProgramGraph.h:158:13: warning: control may reach end of non-void function [-Wreturn-type]
            }
            ^
1 warning generated.
[642/1187] Building CXX object src/storm-pgcl/CMakeFiles/storm-pgcl.dir/builder/ProgramGraphBuilder.cpp.o
In file included from ../src/storm-pgcl/builder/ProgramGraphBuilder.cpp:1:
In file included from ../src/storm-pgcl/builder/ProgramGraphBuilder.h:4:
../src/storm-pgcl/storage/ppg/ProgramGraph.h:158:13: warning: control may reach end of non-void function [-Wreturn-type]
            }
            ^
1 warning generated.
[656/1187] Building CXX object src/storm-pgcl/CMakeFiles/storm-pgcl.dir/storage/ppg/ProgramAction.cpp.o
In file included from ../src/storm-pgcl/storage/ppg/ProgramAction.cpp:2:
../src/storm-pgcl/storage/ppg/ProgramGraph.h:158:13: warning: control may reach end of non-void function [-Wreturn-type]
            }
            ^
1 warning generated.
[657/1187] Building CXX object src/storm-pgcl/CMakeFiles/storm-pgcl.dir/storage/ppg/ProgramEdge.cpp.o
In file included from ../src/storm-pgcl/storage/ppg/ProgramEdge.cpp:2:
../src/storm-pgcl/storage/ppg/ProgramGraph.h:158:13: warning: control may reach end of non-void function [-Wreturn-type]
            }
            ^
1 warning generated.
[659/1187] Building CXX object src/storm-pgcl/CMakeFiles/storm-pgcl.dir/storage/ppg/ProgramEdgeGroup.cpp.o
In file included from ../src/storm-pgcl/storage/ppg/ProgramEdgeGroup.cpp:2:
../src/storm-pgcl/storage/ppg/ProgramGraph.h:158:13: warning: control may reach end of non-void function [-Wreturn-type]
            }
            ^
1 warning generated.
[660/1187] Building CXX object src/storm-pgcl/CMakeFiles/storm-pgcl.dir/storage/ppg/ProgramGraph.cpp.o
In file included from ../src/storm-pgcl/storage/ppg/ProgramGraph.cpp:2:
../src/storm-pgcl/storage/ppg/ProgramGraph.h:158:13: warning: control may reach end of non-void function [-Wreturn-type]
            }
            ^
1 warning generated.
[661/1187] Building CXX object src/storm-pgcl/CMakeFiles/storm-pgcl.dir/storage/ppg/ProgramLocation.cpp.o
In file included from ../src/storm-pgcl/storage/ppg/ProgramLocation.cpp:3:
../src/storm-pgcl/storage/ppg/ProgramGraph.h:158:13: warning: control may reach end of non-void function [-Wreturn-type]
            }
            ^
1 warning generated.
[673/1187] Building CXX object src/storm-pgcl-cli/CMakeFiles/storm-pgcl-cli.dir/storm-pgcl.cpp.o
In file included from ../src/storm-pgcl-cli/storm-pgcl.cpp:10:
In file included from ../src/storm-pgcl/builder/ProgramGraphBuilder.h:4:
../src/storm-pgcl/storage/ppg/ProgramGraph.h:158:13: warning: control may reach end of non-void function [-Wreturn-type]
            }
            ^
1 warning generated.
[886/1187] Building CXX object src/storm/CMakeFiles/storm.dir/storage/expressions/ToRationalNumberVisitor.cpp.o
FAILED: src/storm/CMakeFiles/storm.dir/storage/expressions/ToRationalNumberVisitor.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DBOOST_OPTIONAL_CONFIG_USE_OLD_IMPLEMENTATION_OF_OPTIONAL -DBOOST_RESULT_OF_USE_DECLTYPE -I../src -I../src/storm -I../resources/3rdparty/sparsepp -isystem /usr/local/include -Iinclude -isystem ../resources/3rdparty/l3pp -isystem ../resources/3rdparty/gmm-5.0/include -isystem ../resources/3rdparty/eigen-3.3-beta1 -isystem ../resources/3rdparty/exprtk -isystem ../resources/3rdparty/modernjson/src -isystem resources/3rdparty/cudd-3.0.0/include -isystem ../resources/3rdparty/sylvan/src -isystem ../resources/3rdparty/cpptemplate -isystem /Users/macbook/Code/i2/carl/src -isystem /usr/local/include/eigen3 -isystem /usr/local/include/ginac -mpopcnt -std=c++14 -stdlib=libc++ -ftemplate-depth=1024 -O3 -DNDEBUG -ffast-math -fno-finite-math-only -flto -march=native -fomit-frame-pointer -fPIC -MD -MT src/storm/CMakeFiles/storm.dir/storage/expressions/ToRationalNumberVisitor.cpp.o -MF src/storm/CMakeFiles/storm.dir/storage/expressions/ToRationalNumberVisitor.cpp.o.d -o src/storm/CMakeFiles/storm.dir/storage/expressions/ToRationalNumberVisitor.cpp.o -c ../src/storm/storage/expressions/ToRationalNumberVisitor.cpp
../src/storm/storage/expressions/ToRationalNumberVisitor.cpp:133:39: error: call to 'rationalize' is ambiguous
            return RationalNumberType(carl::rationalize<storm::RationalNumber>(static_cast<size_t>(expression.getValue())));
                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/macbook/Code/i2/carl/src/carl/numbers/adaption_gmpxx/operations.h:225:18: note: candidate function [with T = __gmp_expr<mpq_t, mpq_t>]
inline mpq_class rationalize<mpq_class>(float n) {
                 ^
/Users/macbook/Code/i2/carl/src/carl/numbers/adaption_gmpxx/operations.h:231:18: note: candidate function [with T = __gmp_expr<mpq_t, mpq_t>]
inline mpq_class rationalize<mpq_class>(double n) {
                 ^
/Users/macbook/Code/i2/carl/src/carl/numbers/adaption_gmpxx/operations.h:237:18: note: candidate function [with T = __gmp_expr<mpq_t, mpq_t>]
inline mpq_class rationalize<mpq_class>(int n) {
                 ^
/Users/macbook/Code/i2/carl/src/carl/numbers/adaption_gmpxx/operations.h:242:18: note: candidate function [with T = __gmp_expr<mpq_t, mpq_t>]
inline mpq_class rationalize<mpq_class>(uint n) {
                 ^
/Users/macbook/Code/i2/carl/src/carl/numbers/adaption_gmpxx/operations.h:247:18: note: candidate function [with T = __gmp_expr<mpq_t, mpq_t>]
inline mpq_class rationalize<mpq_class>(sint n) {
                 ^
1 error generated.
[891/1187] Building CXX object src/storm/CMakeFiles/storm.dir/storage/expressions/ExprtkExpressionEvaluator.cpp.o
ninja: build stopped: subcommand failed.
```

Note that CLN & Ginac were installed with Homebrew, not sure if that’s relevant (libstd and so on?).

**Comment by cdehnert** [comment]
At least in the current master this problem should already have been fixed by using carl::uint instead of size_t as the target of the static_cast.

**Comment by tjanson** [comment]
Okay, so storm 1.0.1 (which is the current `stable`) is incompatible with the latest CArL (17.08). Matthias and Sebastian were already aware of this.

Thus this is essentially a documentation issue. I’ll keep this open until storm’s [install instructions](http://www.stormchecker.org/documentation/installation/installation.html#obtaining-the-source-code) work (i.e., `stable` is updated).

**Comment by tjanson** [comment]
Resolved with 1.1.0.
